packages/model-viewer/src/three-components/SmoothControls.ts

```diff
@@ -571,47 +578,90 @@ export class SmoothControls extends EventDispatcher {
     this.userAdjustOrbit(deltaTheta, deltaPhi, 0);
   }
 
-  private onPointerDown = (event: MouseEvent|TouchEvent) => {
-    this.pointerIsDown = true;
-    this.isUserPointing = false;
+  private onPointerDownFactory<E extends MouseEvent|TouchEvent>(
```

Comment:
I too, am wary about making code DRY to a fault. Personally, I think here the factories are warranted especially for `pointerDown` and `pointerUp`. To me it seems intuitive to have the `canInteract` guard and `this.element.style.cursor = 'grab' appear in a single place and the factories allow this to achieved in a consistent style.

I feel a lot less strongly above the `pointerMove` factory, especially since I think the `event.preventDefault()` in that factory is only applicable to `touchmove`. I placed it in a factory because in the previous implementation it was run for both `mouse` and `touch` events and I didn't want to introduce potentially breaking changes which are outside the scope of the PR.

Reply:
If we get rid of the factories, do we opt for shared methods or duplicate code?

Reply:
Personally, I find shared methods to be a bit easier to read and reason about than factories. In debugging iOS, somehow when your `touchMode` is set to `null`, it does stop calling your `pointerMoveFactory`, yet somehow the browser does not start getting touchmove events again. It's as though `preventDefault()` is being remembered somehow. Seems like a Safari bug, but one we'll have to work around. I wonder if your factory somehow confused it? Anyway, try to simplify your change to the  minimum to achieve your goal and let's see if that helps. Should at least make it easier to debug.

Reply:
I imagine this is what you mean, but I want to clarify for anyone else who might come across this PR that `pointerMoveFactory` is **not** called in the `touchmove` event handler. The function _returned_ (or _produced_ if one wants to stick to the factory analogy) by `pointerMoveFactory` is, hence the name.

Thanks for debugging. I think I understand where the issue is. I pushed a fix.

Reply:
Yes, fair enough. I just tested  your change; it definitely had an  effect, but it's still not right. Now it mostly works on iOS, except if the page is scrolled to the top or you're especially quick with your gesture, somehow it ends up doing both scrolling and rotating (so now `preventDefault()` is not being called). I'd recommend removing the `pointerMoveFactory` and see if that helps.

Reply:
Hmm. This is going to be painful to fix without a device to test on.
[Browserstack](https://www.browserstack.com/open-source) has an open source program which would really come in handy here. Is this something you can look into?

Reply:
I simplified as much as possible at the expense of consistency.
You'll notice that `onPointerDown` expects a `fn` parameter. This is necessary otherwise the `canInteract` guard would not terminate the entire event handler, only `onPointerDown`.

Reply:
Thank you! Sadly it behaves  the same. Agreed that debugging this without a repro device  is pretty unreasonable. We in fact have a BrowserStack account (they do  our CI testing), but I'm not sure  how to give you access...

Reply:
I've never used BrowserStack before, but [Live For Teams](https://www.browserstack.com/live-for-teams) seems to be the relevant feature. I imagine that it allows user-specific permissions. Is it part of the open source program?

Reply:
@elalish I revisited this issue and pushed some changes (which I wasn't able to test) on branch `lucadalli:fix-ios-scrolling`.

Could you kindly check if it behaves any differently? Also, what version of iOS are you testing on?

The changes reflect the bugs/fixes reported below:
https://github.com/metafizzy/flickity/issues/457
https://github.com/metafizzy/flickity/blob/master/js/drag.js#L63
https://bugs.webkit.org/show_bug.cgi?id=184250#c10

Reply:
Wow, good find; Webkit strikes again! I just tried your `fix-ios-scrolling` branch and it seems to work great 🥳. I'm testing on an old iPhone SE on 14.7. Go ahead and merge that in here and I'll review. Thanks!

Reply:
Fantastic! 🎉🎉🎉

Branch `fix-ios-scrolling` had two changes:
1. `touchmove` and `mousemove` event listeners attached to `window`
2. the fix suggested in https://github.com/metafizzy/flickity/issues/457#issuecomment-254501356

Since we have a separate issue for attaching `mousemove` to `window` (#2682), I only cherry picked change 2 into this branch.
Kindly confirm everything still works, even with `touchmove` being attached to the element instead of `window`.

Reply:
Okay, that's looking the same as the `fix-ios-scaling` branch. However, upon further testing I now see another less common error (same on both branches). Repro steps (in iOS; I think  I got this to happen once on Android too, but I  was unable to  repro it, but it's 100% on iOS):
1) Start scrolling the page with a finger outside the MV element
2) touch a second finger inside the MV element
3) move both  fingers, note the page both  scrolls and the model zooms (bug 1).
4) remove the first finger, followed by the second finger (order matters).
5) The page is not responsive outside of the  MV element (cannot scroll) until MV is interacted with again (bug 2).

Reply:
Good catch! While I wasn't able to reproduce bug 2 on my Android device, I was able to reproduce bug 1 on both `dynamic-move-handlers` and `master`.
I find it odd that scroll locking (bug 2) is happening on this branch. Unlike `fix-ios-scaling` which attaches `touchmove` to `window`, this branch attaches the `touchmove` listener to the element, so I'm not sure why interaction outside of the element does not produce scrolling.

I pushed a change which fixes bug 1 and supposedly bug 2 as well. It has the added benefit of making the code more elegant too 😀

Reply:
Well done, I can no longer repro either of those bugs in iOS or Android, nor do I notice  any other weird behaviors. 🥳